sdk/src/main/java/com/microsoft/hydralab/performance/PerformanceTestResult.java

```diff
@@ -17,7 +17,7 @@ public class PerformanceTestResult {
     /**
      * TODO: Apply a max size to avoid OOM
      */
-    public List<PerformanceInspectionResult> performanceInspectionResults = new ArrayList<>();
+    public List<PerformanceInspectionResult> performanceInspectionResults = new CopyOnWriteArrayList<>();
```

Comment:
Could you share why you want to use CopyOnWriteArrayList? I can understand that you have a reason for it, so it may be a nice move, just curious about why exactly. Is there a chance we will face a multithreading competition in code?

Reply:
For schedule strategy, the `inspect()` is invoked in separated threads, which add the `PerformanceInspectionResult` to the List. So, we need to use CopyOnWriteArrayList to ensure the thread safety.